### PR TITLE
Domain roll event tests

### PR DESCRIPTION
**Pull request by sankarpn**
Adding tests for checking the DomainRollStarting, DomainRollCompleted and PodCycleStarting events when a domain is rolled after domain resource properties are changed as outlined in this JIRA. https://jira.oraclecorp.com/jira/browse/OWLS-88569

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5204/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 1]
```diff
@@ -492,11 +503,258 @@ public void testDomainK8sEventsProcessingFailed() {
...
+    assertTrue(event.getMessage().contains("LOG_HOME"));
+    assertTrue(event.getMessage().contains("DOMAIN_HOME"));
+
+    event = getEvent(opNamespace, domainNamespace1,
+        domainUid, DOMAIN_ROLL_STARTING, "Normal", timestamp);
+    logger.info(Yaml.dump(event));
+    logger.info("verify the event message contains the domainHome changed message is logged");
+    assertTrue(event.getMessage().contains("domainHome"));
+    assertTrue(event.getMessage().contains("logHome"));
+    //********** I don't see the following message  **********************
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:746]: Is this due to pending issue OWLS-89879.   Add to the test description section.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java [hunk 2]
```diff
@@ -401,11 +495,33 @@ public void testServerPodsRestartByChaningPodSecurityContext() {
     assertTrue(verifyRollingRestartOccurred(podsWithTimeStamps, 1, domainNamespace),
         String.format("Rolling restart failed for domain %s in namespace %s", domainUid, domainNamespace));
 
+    logger.info("verify domain roll starting/pod cycle starting events are logged");
+    //************* I don't see this event logged****************************************
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java:499]: Put the pending  issue to test description.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java [hunk 3]
```diff
@@ -450,12 +566,160 @@ public void testServerPodsRestartByChangingImagePullPolicy() {
...
+  }
+
+  /**
+   * Modify the image on the domain resource.
+   * Verify all pods are restarted and back to ready state.
+   * Verifies that the domain roll starting/pod cycle starting events are logged.
+   */
+  @Test
+  @DisplayName("Check restart of pods after image change")
+  public void testRestartWithImageChange() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java:669]: I remember we had a similar test in some other class, which was failing intermittently. You and Maggie worked together to resolve the issue

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 4]
```diff
@@ -1166,4 +1197,18 @@ public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext con
   @interface AssumeWebLogicImage {
   }
 
+  // Utility method to check event
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:1200]: Moved it to K8sEvents

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 5]
```diff
@@ -492,11 +503,258 @@ public void testDomainK8sEventsProcessingFailed() {
...
+    assertTrue(event.getMessage().contains("SERVER_OUT_IN_POD_LOG"));
+
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_ROLL_COMPLETED, "Normal", timestamp);
+  }
+
+  /**
+   * The test modifies the domainHome and mountPath property and verifies the domain roll starting events are logged.
+   */
+  @Order(12)
+  //@Test
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:661]: removed the test, we cannot test it.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:746]: test is removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java:499]: added.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java (hunk 3 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java:669]: Yes, that won't run if we are running tests in 14.1.*. So I added a similar test here which will run for all wls builds.

**Review by vanajamukkara — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 6]
```diff
@@ -492,11 +503,161 @@ public void testDomainK8sEventsProcessingFailed() {
...
+    assertTrue(event.getMessage().contains("SERVER_OUT_IN_POD_LOG"));
+
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_ROLL_COMPLETED, "Normal", timestamp);
+  }
 
   /**
    * Test DomainDeleted event is logged when domain resource is deleted.
    */
-  @Order(10)
+  @Order(13)
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:660]: I don't think it matters much, but there is no "12" anymore in the test order.

**Review by doxiao — COMMENTED** [review]
The added test cases look  good to me.

Do we have test cases to cover all fields listed in userguide/managing-domains/domain-lifecycle/startup#fields-that-cause-servers-to-be-restarted?

**Review by anpanigr — APPROVED** [review]
LGTM

**Pull request merged**